### PR TITLE
[Merged by Bors] - feat: add missing unit tests to common (vf-000)

### PR DESCRIPTION
**Pull request by effervescentia**
**Fixes or implements VF-000**

### Brief description. What is this change?

Implement missing unit tests to improve coverage

### Checklist

- [x] this is a breaking change and should publish a new major version
- [x] appropriate tests have been written

BREAKING CHANGE: removes unused utility functions

### CODE REVIEW

File: packages/common/src/utils/object/keys.ts [hunk 1]
```diff
@@ -1,68 +1 @@
-import { AnyRecord } from '@common/types';
-import _cloneDeep from 'lodash/cloneDeep';
-import _toPath from 'lodash/toPath';
-
 export const getKeys = <T>(obj: T): (keyof T)[] => Object.keys(obj) as (keyof T)[];
-
-export const getIn = <T, K extends keyof T | string>(rawObj: T, key: K | K[], def?: any, index = 0): rawObj is T & Record<K, unknown> => {
```
Comment (effervescentia) [inline packages/common/src/utils/object/keys.ts:7]: not used in our codebase anymore

File: packages/common/src/utils/object/keys.ts [hunk 2]
```diff
@@ -1,68 +1 @@
...
-  }
-
-  while (obj && i < path.length) {
-    obj = obj[path[i++]];
-  }
-
-  return obj === undefined ? def : obj;
-};
-
-export const setIn = (obj: AnyRecord, path: unknown, value: unknown) => {
```
Comment (effervescentia) [inline packages/common/src/utils/object/keys.ts:23]: not used in our codebase anymore

File: packages/common/src/utils/promise.ts [hunk 3]
```diff
@@ -9,23 +9,3 @@ export const rejectIn = (timeout: number): Promise<never> =>
       reject(new Error('Rejected by timeout!'));
     }, timeout);
   });
-
-interface ControlledPromise<T> extends Promise<T> {
-  reject: (reason?: any) => void;
-  resolve: (value: T) => void;
-}
-
-export const createControlledPromise = <T>(): ControlledPromise<T> => {
```
Comment (effervescentia) [inline packages/common/src/utils/promise.ts:18]: not used in our codebase anymore

File: packages/common/src/utils/string.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 const TAGS_REGEXP = /(<([^>]+)>)/gi;
-const TRAILING_UNDERSCORES_REGEXP = /^_+|_+$/g;
+const TRAILING_UNDERSCORES_REGEXP = /(^_+)|(_+$)/g;
```
Comment (effervescentia) [inline packages/common/src/utils/string.ts:2]: should fix a sonarcloud error that warned about this ambiguous pattern
https://sonarcloud.io/project/issues?issues=AX0phB3lhOt5ghKv_b89&open=AX0phB3lhOt5ghKv_b89&id=voiceflow_libs

File: packages/common/src/utils/object/diff.ts [hunk 5]
```diff
@@ -1,32 +1,8 @@
-import { AnyRecord, DeepPartial } from '@common/types';
+import { AnyRecord } from '@common/types';
 import _transform from 'lodash/transform';
 
-import { isObject } from './common';
-
 const defaultCompare = <T>(left: T, right: T): boolean => left === right;
 
-export const getDiff = <S extends AnyRecord>(object: S, base: S, compare = defaultCompare): DeepPartial<S> => {
```
Comment (effervescentia) [inline packages/common/src/utils/object/diff.ts:8]: not used in our codebase anymore

File: packages/common/src/utils/normalized.ts [hunk 6]
```diff
@@ -109,25 +132,10 @@ export const addAllNormalizedByKeys = <T extends ObjectWithId | unknown, K exten
...
-
+/**
+ * @deprecated prefer `normal-store`
+ */
 export const reorderKeys = <T>(newKeyArray: string[], byKey: Record<string, T>): Normalized<T> => ({
   allKeys: newKeyArray,
   byKey,
 });
-
-export const removeAllNormalizedByKeys = <T>({ allKeys, byKey }: Normalized<T>, targetKeys: string[]): Normalized<T> => {
```
Comment (effervescentia) [inline packages/common/src/utils/normalized.ts:126]: not used in our codebase anymore

File: packages/common/src/utils/normalized.ts [hunk 7]
```diff
@@ -109,25 +132,10 @@ export const addAllNormalizedByKeys = <T extends ObjectWithId | unknown, K exten
   };
 };
 
-export const removeNormalizedByKey: RemoveNormalizedByKey = <T>({ allKeys, byKey }: Normalized<T>, targetKey: string): Normalized<T> => {
```
Comment (effervescentia) [inline packages/common/src/utils/normalized.ts:112]: not used in our codebase anymore

File: packages/common/src/utils/normalized.ts [hunk 8]
```diff
@@ -61,37 +66,55 @@ export const normalize = <T extends ObjectWithId | unknown>(items: T[], getKey?:
...
+/**
+ * @deprecated prefer `normal-store`
+ */
 export const updateNormalizedByKey = <T, N extends Normalized<T>>({ byKey, ...rest }: N, key: string, obj: T) =>
   ({
     ...rest,
     byKey: { ...byKey, [key]: obj },
   } as N);
 
-export const patchNormalizedByKey = <T, N extends Normalized<T>>(normalized: N, key: string, obj: Partial<T>) =>
```
Comment (effervescentia) [inline packages/common/src/utils/normalized.ts:82]: not used in our codebase anymore

File: packages/common/src/utils/normalized.ts [hunk 9]
```diff
@@ -61,37 +66,55 @@ export const normalize = <T extends ObjectWithId | unknown>(items: T[], getKey?:
...
+ */
 export const safeGetNormalizedByKey: SafeGetNormalizedByKey = <T>({ byKey }: Normalized<T>, key: string) =>
   hasProperty(byKey, key) ? byKey[key] : null;
 
+/**
+ * @deprecated prefer `normal-store`
+ */
 export const getAllNormalizedByKeys = <T>({ byKey }: Normalized<T>, keys: string[]) => keys.map((key) => byKey[key]);
 
-export const safeGetAllNormalizedByKeys = <T>({ allKeys, byKey }: Normalized<T>, keys: string[]): T[] =>
```
Comment (effervescentia) [inline packages/common/src/utils/normalized.ts:73]: not used in our codebase anymore

**Review by trs — APPROVED** [review]

**Pull request closed without merging**